packages/manager/src/features/CloudPulse/Alerts/CreateAlert/GeneralInformation/ResourceMultiSelect.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import { Controller, useFormContext } from 'react-hook-form';
+
+import { Autocomplete } from 'src/components/Autocomplete/Autocomplete';
+import { useResourcesQuery } from 'src/queries/cloudpulse/resources';
+
+import type { Item } from '../../constants';
+import type { CreateAlertDefinitionForm } from '../types';
+import type { AlertServiceType } from '@linode/api-v4';
+import type { FieldPathByValue } from 'react-hook-form';
+
+interface CloudPulseResourceSelectProps {
+  /**
+   * engine option type selected by the user
+   */
+  engine: null | string;
+  /**
+   * name used for the component to set in the form
+   */
+  name: FieldPathByValue<CreateAlertDefinitionForm, string[]>;
+  /**
+   * region selected by the user
+   */
+  region: string | undefined;
+  /**
+   * service type selected by the user
+   */
+  serviceType: AlertServiceType | null;
+}
+
+export const CloudPulseMultiResourceSelect = (
+  props: CloudPulseResourceSelectProps
+) => {
+  const { engine, name, region, serviceType } = { ...props };
+  const { control, setValue } = useFormContext<CreateAlertDefinitionForm>();
+
+  const { data: resources, isError, isLoading } = useResourcesQuery(
+    Boolean(region && serviceType),
+    serviceType?.toString(),
+    {},
+    engine !== null ? { engine, region } : { region }
+  );
+
+  const getResourcesList = React.useMemo((): Item<string, string>[] => {
+    return resources && resources.length > 0
+      ? resources.map((resource) => ({
+          label: resource.label,
+          value: resource.id,
+        }))
+      : [];
+  }, [resources]);
+
+  React.useEffect(() => {
+    setValue(name, []);
+  }, [region, serviceType, engine, setValue, name]);
+
+  return (
+    <Controller
+      render={({ field, fieldState }) => (
+        <Autocomplete
+          errorText={
+            fieldState.error?.message ??
+            (isError ? 'Failed to fetch the resources.' : '')
+          }
+          onChange={(_, resources: { label: string; value: string }[]) => {
+            const resourceIds = resources.map((resource) => resource.value);
+            field.onChange(resourceIds);
```

Comment:
I believe we can do this without a useEffect, please double check this.
```suggestion
            setValue(name, resourceIds);
```

Reply:
@jaalah-akamai , why use setValue here instead of field.onChange(). I will double check the useEffect usecase

Reply:
As discussed, validation is done with form itself we don't need an additional trigger when values are updating. So going ahead with `field.onChange` and added a comment in the code explaining why we are using the `useEffect()`.